src/components/Input/__tests__/Input.spec.tsx

```diff
@@ -18,29 +28,77 @@ describe('<Input> that renders an input field', () => {
     });
 
     it('should add classes when props are changed', () => {
-        view = render(<Input context="critical" size="large" isBlock disabled />);
+        view = renderComponent({
+            context: 'critical',
+            isBlock: true,
+            disabled: true,
+        });
         const textbox = screen.getByRole('textbox');
 
         expect(view.container).toMatchSnapshot();
         expect(textbox).toBeInTheDocument();
         expect(textbox).toHaveClass(
-            'Input Input--context_critical Input--size_large Input--isBlock'
+            'Input Input--context_critical Input--size_medium Input--isBlock'
         );
         expect(textbox).toHaveAttribute('disabled');
     });
 
+    it('should render input with label', () => {
+        const label = 'Input Label';
+        view = renderComponent({ label });
+        const labelText = screen.getByText(label);
+
+        expect(labelText).toBeInTheDocument();
+        expect(labelText).toHaveAttribute('for', 'input-field');
```

Comment:
not sure if it adds value, but could also test for `expect(screen.getByLabelText(label)).toBeInTheDocument()`.